prov/gni/src/gnix_nic.c

```diff
@@ -1253,6 +1253,10 @@ int gnix_nic_alloc(struct gnix_fid_domain *domain,
 		GNIX_INFO(FI_LOG_EP_CTRL, "Allocated NIC:%p\n", nic);
 	}
 
+	if (nic) {
+		nic->requires_lock = domain->thread_model != FI_THREAD_COMPLETION;
+	}
```

Comment:
I'm not convinced this is correct.  It could be possible in the current nic allocation scheme for a nic to be associated with ep's bound to different completion queues.  We could change the allocation scheme to require a nic to be allocated per endpoint in the case of ```FI_THREAD_COMPLETION``` being requested.